### ISSUE TITLE
`post-comment` not working?

### ISSUE THREAD

**Issue by ospfranco** [tangro/actions-test#3]
Hey, thanks for the step, but I tried the post-comment option and nothing got posted to one of my PRs.

My config:
```
jobs:
  tests:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Test using Node.js
        uses: actions/setup-node@v1
        with:
          node-version: '12'
      - name: Install Dependencies
        run: yarn
      - name: Run tests
        uses: tangro/actions-test@v1.1.6
        with:
          post-comment: true
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GITHUB_CONTEXT: ${{ toJson(github) }}
```

Any help would be appreciated, thanks!

Edit: I have to say our test set is not exiting cleanly, we have some open handles that cause an exit with 1, could this be part of the problem?

**Comment by ospfranco** [comment]
lol, I just realized that even if the tests don't fail it is this script that fails... any support?

**Comment by ankri** [comment]
Did you check that you have a `test:ci` with `jest --testLocationInResults --ci --outputFile=test_results.json --json` in your `package.json` -> [Documentation](https://github.com/tangro/actions-test#usage)

Otherwise please post the "Raw Logs" of the failing workflow

![image](https://user-images.githubusercontent.com/2842920/100920534-8d6ac900-34db-11eb-9d51-a7aaa3043430.png)

**Comment by ospfranco** [comment]
yes, the command is correct, I forked the project and trying to add debug output and it seems it cannot require the `test_results.json` file, it complains it cannot find it

**Comment by ospfranco** [comment]
Sorry, previous comment is not valid, I was doing some other testing, I added some debug output on the wrapWithSetStatus function's catch block and I get this error:

`ReferenceError: name is not defined`

**Comment by ankri** [comment]
Let me know what you found out. I have created a simple reproduction repo, that works:

**package.json**
```json
"scripts": {
  "test": "jest",
  "test:ci": "jest --testLocationInResults --ci --outputFile=test_results.json --json"
},
"devDependencies": {
  "jest": "26.6.3"
}
```

**test.js**
```js
describe("Test suite", () => {
  it("should fail", () => {
    expect(false).toBe(true);
  });
});
```

**.github/workflows/test.yml**
```yml
name: Test
on: [push]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout latest code
        uses: actions/checkout@v2
      - name: Use Node.js 12.x
        uses: actions/setup-node@v2.1.2
        with:
          node-version: 12.x
      - name: Run npm install
        run: npm install
      - name: Run tests
        uses: tangro/actions-test@v1.1.6
        with:
          post-comment: true
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GITHUB_CONTEXT: ${{ toJson(github) }}
```

**Comment by ankri** [comment]
> `ReferenceError: name is not defined`

Do you have some more information? Maybe the log file that the error happened in? Without context it is a bit hard to follow which `name` is supposed to be not defined.

**Comment by ospfranco** [comment]
that is what I'm trying to figure out, but the logs show no error because all the errors are swallowed, trying to get the stack trace currently, I'll let you know what I find

**Comment by ospfranco** [comment]
yeah, it seems to be working on a simple empty project, but on our big project it is not... weird, somewhere a name variable is not being set

**Comment by ankri** [comment]
Maybe remove the `checkRun` stuff in your fork. You don't need it for the comment.

**Comment by ankri** [comment]
just this
```typescript
export async function createChecksFromTestResults({
  pathToTestOutput,
  context
}: {
  pathToTestOutput: string;
  context: GitHubContext<{}>;
}): Promise<FormattedTestResults> {
  const [owner, repo] = context.repository.split('/');

  const formattedTestResults = require(pathToTestOutput) as FormattedTestResults;

  return formattedTestResults;
}

```

**Comment by ospfranco** [comment]
I've managed to narrow it down to the `getCheckRunForAction` function, but not sure exactly what is failing

**Comment by ankri** [comment]
Does your workflow have a name?

```yml
name: Test <--------------
on: [push]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout latest code
        uses: actions/checkout@v2
      - name: Use Node.js 12.x
        uses: actions/setup-node@v2.1.2
        with:
          node-version: 12.x
      - name: Run npm install
        run: npm install
      - name: Run tests
        uses: tangro/actions-test@v1.1.6
        with:
          post-comment: true
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GITHUB_CONTEXT: ${{ toJson(github) }}

```

**Comment by ospfranco** [comment]
yes
```yaml

name: JS Checks
on: [pull_request]

jobs:
  tests:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Use Node.js 12.x
        uses: actions/setup-node@v2.1.2
        with:
          node-version: 12.x
      - name: Install Dependencies
        run: yarn
      - name: Run tests
        uses: ospfranco/gh-jester@v1.0.9
        with:
          post-comment: true
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GITHUB_CONTEXT: ${{ toJson(github) }}
    
  flow:
    runs-on: ubuntu-latest

    steps:
      - uses : actions/checkout@v2
      - name: Test using Node.js
        uses: actions/setup-node@v1
        with:
          node-version: '12'
      - name: Install Dependencies
        run: yarn
      - name: Flow
        run: yarn flow

  lint:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Test using Node.js
        uses: actions/setup-node@v1
        with:
          node-version: '12'
      - name: Install Dependencies
        run: yarn
      - name: Lint
        run: yarn lint

```

**Comment by ospfranco** [comment]
```    
"data": {
        "total_count": 0,
        "check_runs": []
    }
```
this is the data field of the checkRunsResponse request

**Comment by ospfranco** [comment]
commenting this lines:

```javascript
// const checkRun = await getCheckRunForAction({
  //   context
  // });
```

and

```javascript
// await updateCheckRun({
    //   context,
    //   checkRunId: checkRun.id,
    //   name: checkRun.name || 'NO_NAME',
    //   checks
    // });
```

Does let the action finish, but no comment is posted...

**Comment by ospfranco** [comment]
added some extra debug, after removing the check function the code reaches the message posting via the GitHub object, but nothing is posted, I'm guessing because the checkRun has not been updated?

**Comment by ankri** [comment]
No. The comment will be posted to the `context` (e.g. the commit/PR). It should not be influenced by the Check Run.

I am guessing that testSummary is null/undefined on line 57:
https://github.com/tangro/actions-test/blob/4b95861611c75b8e3db0d007c52e8e044a9f244f/src/main.ts#L57-L61

**Comment by ospfranco** [comment]
no, it is not, I added debug out put there as well and it reaches the create comment function.

Just to make my life easier, how do you test this? I've been creating releases manually, but this is super slow, is there any easier way to do this?

**Comment by ospfranco** [comment]
Could it be that the repo I'm trying to make this work is private?

**Comment by ankri** [comment]
> Just to make my life easier, how do you test this? I've been creating releases manually, but this is super slow, is there any easier way to do this?

I sadly do not know any other way to test GitHub Actions. But we did all our actions during the beta and haven't checked up with the documentation whether the best practices changed or they added stuff to their documentation.

But I remember that there are debug flags you can set. But they are sadly buried in the docs somewhere.

> no, it is not, I added debug out put there as well and it reaches the create comment function.
Since we do not have problems with that in our repositories and we both have working test repositories it has to be either your fork or your workflow you are using the action in.

The code for posting the comment is fairly simple since it's just a wrapper around the official github octokit.

> Could it be that the repo I'm trying to make this work is private?

No, all our repositories are private. The actions are public because actions have to be public.

---

I'm sorry. I do not know why it's not working with your fork. When the `context` ist correct and the `GITHUB_TOKEN` is set "everything should work"™️. Otherwise it might be an error in the Octokit but I doubt that.

**Comment by ospfranco** [comment]
<img width="919" alt="Screenshot 2020-12-03 at 09 52 52" src="https://user-images.githubusercontent.com/1634213/100986463-8381ae00-354d-11eb-981b-7e880d3c4936.png">

Well I finally got it working, the create comment was using `context.sha` which for some reason on our repo does not match with the specific commit (?) by using ref_head at least the comment is posted in our PR, it's probably going to cause some trouble with subsequent commits, but I'm happy to see some output finally

**Comment by ospfranco** [comment]
Well, found out why the SHA is different, the trigger I was using is [pull_request] not push, therefore the SHA is not of the commit but of the PR maybe? changing the trigger to push makes things work again. closing this.
